components/service/sync-autofill/src/main/java/mozilla/components/service/sync/autofill/GeckoCreditCardsAddressesStorageDelegate.kt

```diff
@@ -19,7 +20,8 @@ import mozilla.components.concept.storage.CreditCardsAddressesStorageDelegate
  */
 class GeckoCreditCardsAddressesStorageDelegate(
     private val storage: Lazy<CreditCardsAddressesStorage>,
-    private val scope: CoroutineScope = CoroutineScope(Dispatchers.IO)
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.IO),
+    private val isCCAutofillEnabled: () -> Boolean = { false }
```

Comment:
I am not quite sure if we want to add this parameter to `GeckoCreditCardsAddressesStorageDelegate`. I think it should be enough if we just intercept the request and block it from showing.

Reply:
We can disable this in `PromptFeature`. And this condition is indeed checked there also.
Here it's used to skip querying storage for data that won't be used.

Reply:
This is also the same approach initially used for toggling logins autofill before there was a `GeckoRuntimeSettings` available for it.
I can remove this if you want but I feel like there's a real benefit in having this check also here.